app/src/main/java/com/example/fuji/CreateBoardFragment.kt

```diff
@@ -28,9 +59,44 @@ class CreateBoardFragment : Fragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        view.findViewById<ImageView>(R.id.back_image).setOnClickListener {
-            var Board: BoardsActivity = activity as BoardsActivity
-            Board.switchToBoardFragment()
+        val boardNameInputView = view.findViewById<EditText>(R.id.create_board_entry)
+
+        view.findViewById<ImageView>(R.id.create_board_back_button).setOnClickListener {
+            var board: BoardsActivity = activity as BoardsActivity
+            board.switchToBoardFragment()
+        }
+
+        view.findViewById<Button>(R.id.create_board_button).setOnClickListener {
+            val boardNameInput = boardNameInputView.text.toString().trim()
+
+            if (boardNameInput.isEmpty()) {
+                Toast.makeText(context, "Please give the board a name first", Toast.LENGTH_SHORT).show()
+                return@setOnClickListener
+            }
+            if (boardNameInput.length > 20) {
+                Toast.makeText(context, "Please give a board name of twenty characters or less", Toast.LENGTH_SHORT).show()
+                return@setOnClickListener
+            }
+            // [FIREBASE ADD BOARD BEGIN]
+            addBoard(boardNameInput)
+            // [FIREBASE ADD BOARD END]
+            var board: BoardsActivity = activity as BoardsActivity
+            Handler().postDelayed({board.switchToBoardFragment()}, 3000)
         }
     }
-}
\ No newline at end of file
+}
+
+/*
+val newBoard = hashMapOf(
+                "Title" to "New Board",
+                "Name" to "Board Name"
+        )
+
+        db.collection("test]").add(newBoard).addOnSuccessListener { documentReference ->
+            Log.d(TAG, "DocumentSnapshot added with ID: ${documentReference.id}")
+            Toast.makeText(context, "Board Created", Toast.LENGTH_SHORT).show()
+        }.addOnFailureListener { e ->
+            Log.w(TAG, "Error adding Board", e)
+            Toast.makeText(context, "Board Creation Failed", Toast.LENGTH_SHORT).show()
+        }
+ */
```

Comment:
best practice is to remove commented code

Reply:
Deleted